File: app/src/main/java/cdreyfus/xebia_henri_potier/basket/Percentage.java
package cdreyfus.xebia_henri_potier.basket;

import cdreyfus.xebia_henri_potier.basket.CommercialOffer;

public class Percentage extends CommercialOffer {
    private  int percentage;

    public Percentage(int percentage) {
        this.type = "percentage";
        this.percentage = percentage;
    }

    public float applyPercentage(float basketValue){
        return basketValue * (1 - (float) percentage/100);

    }

    public int getValue() {
        return percentage;
    }

    public void setPercentage(int percentage) {
        this.percentage = percentage;
    }

    @Override
    public float applyOffer(float basketValue) {
        return applyPercentage(basketValue);
    }
}

File: app/src/main/java/cdreyfus/xebia_henri_potier/basket/promotion/Percentage.kt
package cdreyfus.xebia_henri_potier.basket.promotion

class Percentage(percentage: Int) : CommercialOffer() {
    var value: Int = 0
        private set

    init {
        this.type = "percentage"
        this.value = percentage
    }

    fun applyPercentage(basketValue: Float): Float {
        return basketValue * (1 - value.toFloat() / 100)

    }

    fun setPercentage(percentage: Int) {
        this.value = percentage
    }

    override fun applyOffer(basketValue: Float): Float {
        return applyPercentage(basketValue)
    }
}
